core/dbt/parser/schemas.py

```diff
@@ -816,6 +816,24 @@ def parse_patch(
         source_file: SchemaSourceFile = self.yaml.file
         if patch.yaml_key in ['models', 'seeds', 'snapshots']:
             unique_id = self.manifest.ref_lookup.get_unique_id(patch.name, None)
+            if unique_id:
+                resource_type = unique_id.split('.')[0]
+                if (unique_id.split('.')[0] + 's') != patch.yaml_key:
+                    warn_or_error(
+                        f"WARNING: {patch.name} is a {resource_type} node, "
+                        f"but it is specified in the {patch.yaml_key} section "
+                        f"of {patch.original_file_path}. "
+                        f"To fix this warning, place the `{patch.name}` specification "
+                        f"under the {resource_type} key instead.")
+
+                    return
+
+            else:
+                if not self.manifest.find_disabled_by_name(patch.name, None):
+                    # need to raise an exception when a refable node does not
+                    # have a unique id and is not disabled
+                    # todo: do we still want to do this?
+                    raise raise_no_unique_id(patch)
```

Comment:
I don't believe we raise an exception for this today, so I don't think we should start. Unfortunately, this is likely to break a lot of projects, and prevent them from upgrading to v1.0 :(